libsodium 1.0.18 does not compile with the latest ESP-IDF

Compiling the libsodium version 1.0.18 with ESP-IDF verison `v5.0-dev-1925-g7af86bb335` (latest version at the moment) produces the following errors:

```
[990/1115] Building C object esp-idf/espressif__libsodium/CMakeFiles/__idf_espressif__libsodium.dir/port/randombytes_esp32.c.obj
FAILED: esp-idf/espressif__libsodium/CMakeFiles/__idf_espressif__libsodium.dir/port/randombytes_esp32.c.obj 
/home/b1ack/.espressif/tools/xtensa-esp32s2-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32s2-elf/bin/xtensa-esp32s2-elf-gcc -DCONFIGURED -DHAVE_WEAK_SYMBOLS -DHUB_REVISION=\"E4\" -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -DNATIVE_LITTLE_ENDIAN -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/home/b1ack/projects/abra/esp-firmware/build/config -I/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/libsodium/src/libsodium/include -I/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/port_include -I/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/libsodium/src/libsodium/include/sodium -I/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/port_include/sodium -I/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/port -I/home/b1ack/projects/esp-idf/components/newlib/platform_include -I/home/b1ack/projects/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/b1ack/projects/esp-idf/components/freertos/esp_additions/include/freertos -I/home/b1ack/projects/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/b1ack/projects/esp-idf/components/freertos/esp_additions/include -I/home/b1ack/projects/esp-idf/components/esp_hw_support/include -I/home/b1ack/projects/esp-idf/components/esp_hw_support/include/soc -I/home/b1ack/projects/esp-idf/components/esp_hw_support/include/soc/esp32s2 -I/home/b1ack/projects/esp-idf/components/esp_hw_support/port/esp32s2/. -I/home/b1ack/projects/esp-idf/components/esp_hw_support/port/esp32s2/private_include -I/home/b1ack/projects/esp-idf/components/heap/include -I/home/b1ack/projects/esp-idf/components/log/include -I/home/b1ack/projects/esp-idf/components/lwip/include/apps -I/home/b1ack/projects/esp-idf/components/lwip/include/apps/sntp -I/home/b1ack/projects/esp-idf/components/lwip/lwip/src/include -I/home/b1ack/projects/esp-idf/components/lwip/port/esp32/include -I/home/b1ack/projects/esp-idf/components/lwip/port/esp32/include/arch -I/home/b1ack/projects/esp-idf/components/soc/include -I/home/b1ack/projects/esp-idf/components/soc/esp32s2/. -I/home/b1ack/projects/esp-idf/components/soc/esp32s2/include -I/home/b1ack/projects/esp-idf/components/hal/esp32s2/include -I/home/b1ack/projects/esp-idf/components/hal/include -I/home/b1ack/projects/esp-idf/components/hal/platform_port/include -I/home/b1ack/projects/esp-idf/components/esp_rom/include -I/home/b1ack/projects/esp-idf/components/esp_rom/include/esp32s2 -I/home/b1ack/projects/esp-idf/components/esp_rom/esp32s2 -I/home/b1ack/projects/esp-idf/components/esp_common/include -I/home/b1ack/projects/esp-idf/components/esp_system/include -I/home/b1ack/projects/esp-idf/components/esp_system/port/soc -I/home/b1ack/projects/esp-idf/components/esp_system/port/include/private -I/home/b1ack/projects/esp-idf/components/xtensa/include -I/home/b1ack/projects/esp-idf/components/xtensa/esp32s2/include -I/home/b1ack/projects/esp-idf/components/efuse/include -I/home/b1ack/projects/esp-idf/components/efuse/esp32s2/include -I/home/b1ack/projects/esp-idf/components/vfs/include -I/home/b1ack/projects/esp-idf/components/esp_wifi/include -I/home/b1ack/projects/esp-idf/components/esp_event/include -I/home/b1ack/projects/esp-idf/components/esp_netif/include -I/home/b1ack/projects/esp-idf/components/esp_eth/include -I/home/b1ack/projects/esp-idf/components/tcpip_adapter/include -I/home/b1ack/projects/esp-idf/components/esp_phy/include -I/home/b1ack/projects/esp-idf/components/esp_phy/esp32s2/include -I/home/b1ack/projects/esp-idf/components/esp_timer/include -I/home/b1ack/projects/esp-idf/components/mbedtls/port/include -I/home/b1ack/projects/esp-idf/components/mbedtls/mbedtls/include -I/home/b1ack/projects/esp-idf/components/mbedtls/esp_crt_bundle/include -I/home/b1ack/projects/esp-idf/components/mbedtls/mbedtls/library -I/home/b1ack/projects/esp-idf/components/driver/include -I/home/b1ack/projects/esp-idf/components/driver/deprecated -I/home/b1ack/projects/esp-idf/components/driver/esp32s2/include -I/home/b1ack/projects/esp-idf/components/esp_pm/include -I/home/b1ack/projects/esp-idf/components/esp_ringbuf/include -mlongcalls  -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -ggdb -Og -fmacro-prefix-map=/home/b1ack/projects/abra/esp-firmware=. -fmacro-prefix-map=/home/b1ack/projects/esp-idf=/IDF -fstrict-volatile-bitfields -Wno-error=unused-but-set-variable -fno-jump-tables -fno-tree-switch-conversion -std=gnu99 -Wno-old-style-declaration -D_GNU_SOURCE -DIDF_VER=\"v5.0-dev-1925-g7af86bb335\" -DESP_PLATFORM -D_POSIX_READER_WRITER_LOCKS -Wno-unused-function -MD -MT esp-idf/espressif__libsodium/CMakeFiles/__idf_espressif__libsodium.dir/port/randombytes_esp32.c.obj -MF esp-idf/espressif__libsodium/CMakeFiles/__idf_espressif__libsodium.dir/port/randombytes_esp32.c.obj.d -o esp-idf/espressif__libsodium/CMakeFiles/__idf_espressif__libsodium.dir/port/randombytes_esp32.c.obj -c /home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/port/randombytes_esp32.c
/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/port/randombytes_esp32.c:23:15: error: 'esp_random' undeclared here (not in a function); did you mean 'srandom'?
     .random = esp_random,
               ^~~~~~~~~~
               srandom
/home/b1ack/projects/abra/esp-firmware/managed_components/espressif__libsodium/port/randombytes_esp32.c:26:12: error: 'esp_fill_random' undeclared here (not in a function)
     .buf = esp_fill_random,
            ^~~~~~~~~~~~~~~
```

The `idf_component.yml` is the following:

```yml
## IDF Component Manager Manifest File
dependencies:
  espressif/json_parser: "==1.0.0"
  espressif/json_generator: "==1.1.0"
  espressif/esp_modem: "==0.1.13"
  espressif/libsodium: "==1.0.18"
  ## Required IDF version
  idf:
    version: ">=4.1.0"
  # # Put list of dependencies here
  # # For components maintained by Espressif:
  # component: "~1.0.0"
  # # For 3rd party components:
  # username/component: ">=1.0.0,<2.0.0"
  # username2/component2:
  #   version: "~1.0.0"
  #   # For transient dependencies `public` flag can be set.
  #   # `public` flag doesn't have an effect dependencies of the `main` component.
  #   # All dependencies of `main` are public by default.
  #   public: true
```

The toolchain is `xtensa-esp32s2-elf-gcc (crosstool-NG esp-2021r2-patch3) 8.4.0`.

--- a/libsodium/idf_component.yml
+++ b/libsodium/idf_component.yml
@@ -1,6 +1,6 @@
 
-version: "1.0.18"
-description: libsodium port to ESP32
+version: "1.0.19"
+description: libsodium port to ESP
 url: https://github.com/espressif/idf-extra-components/tree/master/libsodium
 dependencies:
   idf: ">=4.2"

